server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSource.java

```diff
@@ -507,8 +510,49 @@ public SortedBinaryDocValues bytesValues(LeafReaderContext context) {
                 return indexFieldData.load(context).getBytesValues();
             }
 
-            public org.elasticsearch.index.fielddata.MultiGeoPointValues geoPointValues(LeafReaderContext context) {
-                return indexFieldData.load(context).getGeoPointValues();
+            public MultiGeoValues geoValues(LeafReaderContext context) {
+                return indexFieldData.load(context).getGeoValues();
+            }
+        }
+    }
+
+    public abstract static class GeoShape extends ValuesSource implements Geo {
```

Comment:
Do we need both `GeoPoint` and `GeoShape` if they are both returning the new `MultiGeoValues`?  I suspect we do need the distinction, but don't know enough about the geo stuff to say.

Reply:
yeah... this is because of the `ValuesSourceConfig#originalValuesSource()`, which depends on the right ValuesSource that knows how to convert the atomic field data from the correct mapped field type: GeoPoint or Geoshape into the MultiGeoValues